### PR TITLE
ci: add developer option for trace fixtures for simulating streaming span

### PR DESCRIPTION
**Pull request by RogerHYang**

### CODE REVIEW

File: src/phoenix/server/main.py [hunk 1]
```diff
@@ -104,22 +119,27 @@ def _get_pid_file() -> Path:
...
                 ),
             ),
-        ):
-            traces.put(span)
+        )
+        Thread(
+            target=_load_spans,
+            args=(traces, fixture_spans, simulate_streaming),
+            daemon=True,
+        ).start()
```
Comment (mikeldking) [inline src/phoenix/server/main.py:142]: Does running as a thread have any impact resource wise in colab? Just checking.

File: src/phoenix/server/main.py [hunk 2]
```diff
@@ -46,10 +48,22 @@ def _get_pid_file() -> Path:
...
 
 
+def _load_spans(
+    traces: Traces,
+    spans: Iterable[Span],
+    simulate_streaming: Optional[bool] = False,
+) -> None:
+    for span in spans:
+        if simulate_streaming:
+            sleep(random())
```
Comment (mikeldking) [inline src/phoenix/server/main.py:58]: random is between 0 to 1 seconds?

File: src/phoenix/server/main.py [hunk 3]
```diff
@@ -46,10 +48,22 @@ def _get_pid_file() -> Path:
...
+        traces.put(span)
+
+
 if __name__ == "__main__":
     primary_dataset_name: str
     reference_dataset_name: Optional[str]
     trace_dataset_name: Optional[str] = None
+    simulate_streaming: Optional[bool] = None
```
Comment (mikeldking) [inline src/phoenix/server/main.py:66]:
```suggestion
    simulate_streaming: Optional[bool] = False
```
Can't we also just have this as the fixture parser default?

**Review by mikeldking — APPROVED** [review]

File: src/phoenix/server/main.py (hunk 1 above)
Comment (RogerHYang) [reply src/phoenix/server/main.py:142]: There shouldn't be. Incidentally, this code path only gets run by `ProcessServer` which spawns a process and this thread will be inside that process.

File: src/phoenix/server/main.py (hunk 2 above)
Comment (RogerHYang) [reply src/phoenix/server/main.py:58]: yes

File: src/phoenix/server/main.py (hunk 3 above)
Comment (RogerHYang) [reply src/phoenix/server/main.py:66]: I was only following the convention from the line prior. Ether way it makes no difference, and we don't need to make an exception just for 'bool'.

File: src/phoenix/server/main.py (hunk 1 above)
Comment (RogerHYang) [reply src/phoenix/server/main.py:150]: Since it's an iterator, we can also put `cycle()` on it and make it infinite...

**Pull request merged**